plugins/catalog-backend/src/next/database/DefaultProcessingDatabase.ts

```diff
@@ -508,6 +518,61 @@ export class DefaultProcessingDatabase implements ProcessingDatabase {
     };
   }
 
+  async listAncestors(
+    txOpaque: Transaction,
+    options: ListAncestorsOptions,
+  ): Promise<ListAncestorsResult> {
+    const tx = txOpaque as Knex.Transaction;
+    const { entityRef } = options;
+    const entityRefs = new Array<string>();
+
+    let currentRef = entityRef;
+    let depth = 0;
+    for (;;) {
+      if (depth++ > MAX_ANCESTOR_DEPTH) {
+        throw new Error(
+          `Unable receive ancestors for ${entityRef}, reached maximum depth of ${MAX_ANCESTOR_DEPTH}`,
+        );
+      }
+
+      const rows = await tx<DbRefreshStateReferencesRow>(
+        'refresh_state_references',
+      )
+        .where({ target_entity_ref: currentRef })
+        .select();
+
+      if (rows.length === 0) {
+        if (depth === 1) {
+          throw new NotFoundError(`Entity ${currentRef} not found`);
+        }
+        throw new NotFoundError(
+          `Entity ${entityRef} has a broken parent reference chain at ${currentRef}`,
+        );
+      }
+
+      const parentRef = rows.find(r => r.source_entity_ref)?.source_entity_ref;
+      if (!parentRef) {
+        // We've reached the top of the tree which is the entityProvider.
+        // In this case we refresh the entity itself.
+        return { entityRefs };
+      }
+      entityRefs.push(parentRef);
+      currentRef = parentRef;
+    }
+  }
+
+  async refresh(txOpaque: Transaction, options: RefreshOptions): Promise<void> {
+    const tx = txOpaque as Knex.Transaction;
+    const { entityRef } = options;
+
+    const updateResult = await tx<DbRefreshStateRow>('refresh_state')
+      .where({ entity_ref: entityRef })
```

Comment:
```suggestion
      .where({ entity_ref: entityRef.toLocaleLowerCase('en-US') })
```

since this is untrusted user input being matched with a strict machine generated value

Reply:
I would say that its best to lowercase it in the routing layer tho